src/library/psi_library.jl

```diff
@@ -1149,3 +1149,415 @@ function build_two_zone_5_bus(; kwargs...)
     end
     return sys
 end
+
+const COST_PERTURBATION_NOISE = rand(1_000_000)
+
+function _duplicate_system(main_sys::PSY.System, twin_sys::PSY.System, HVDC_line::Bool)
+    names = [
+        "114_SYNC_COND_1",
+        "314_SYNC_COND_1",
+        "313_STORAGE_1",
+        "214_SYNC_COND_1",
+        "212_CSP_1",
+    ]
+    for sys in [main_sys, twin_sys]
+        for d in get_components(
+            x -> get_fuel(x) == ThermalFuels.DISTILLATE_FUEL_OIL,
+            ThermalStandard,
+            sys,
+        )
+            for s in get_services(d)
+                remove_service!(d, s)
+            end
+            remove_component!(sys, d)
+        end
+        for d in PSY.get_components(x -> x.name ∈ names, PSY.Generator, sys)
+            for s in get_services(d)
+                remove_service!(d, s)
+            end
+            remove_component!(sys, d)
+        end
+        for d in
+            get_components(x -> get_fuel(x) == ThermalFuels.NUCLEAR, ThermalStandard, sys)
+            set_must_run!(d, true)
+        end
+    end
+
+    # clear time series
+    PSY.clear_time_series!(twin_sys)
+
+    # change names of the systems
+    PSY.set_name!(main_sys, "main")
+    PSY.set_name!(twin_sys, "twin")
+
+    # change the names of the areas and loadzones first
+    for type_ in [PSY.Area, PSY.LoadZone]
+        for b in PSY.get_components(type_, twin_sys)
+            name_ = PSY.get_name(b)
+            main_comp = PSY.get_component(type_, main_sys, name_)
+            # remove the component
+            PSY.remove_component!(twin_sys, b)
+            # change name
+            PSY.set_name!(b, name_ * "_twin")
+            # define time series container
+            IS.assign_new_uuid!(b)
+            # add comopnent to the new sys (main)
+            PSY.add_component!(main_sys, b)
+            # check if it has timeseries
+            if PSY.has_time_series(main_comp)
+                PSY.copy_time_series!(b, main_comp)
+            end
+        end
+    end
+
+    # now add the buses
+    for b in PSY.get_components(PSY.ACBus, twin_sys)
+        name_ = PSY.get_name(b)
+        main_comp = PSY.get_component(PSY.ACBus, main_sys, name_)
+        # remove the component
+        PSY.remove_component!(twin_sys, b)
+        # change name
+        PSY.set_name!(b, name_ * "_twin")
+        # change area
+        PSY.set_area!(
+            b,
+            PSY.get_component(
+                Area,
+                main_sys,
+                PSY.get_name(PSY.get_area(main_comp)) * "_twin",
+            ),
+        )
+        # change number
+        PSY.set_number!(b, PSY.get_number(b) + 10000)
```

Comment:
This is likely a longstanding problem. Bus numbers are stored in the system. If you change it like this, problems will occur with the functions that rely on `sys.bus_numbers`.

Reply:
Should we open an issue in PSY to comment on this?